x-pack/plugins/uptime/server/plugin.ts

```diff
@@ -4,30 +4,92 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-
+import { once } from 'lodash';
 import {
   PluginInitializerContext,
   CoreStart,
   CoreSetup,
   Plugin as PluginType,
   ISavedObjectsRepository,
 } from '../../../../src/core/server';
+import { uptimeRuleFieldMap } from '../common/rules/uptime_rule_field_map';
 import { initServerWithKibana } from './kibana.index';
 import { KibanaTelemetryAdapter, UptimeCorePlugins } from './lib/adapters';
 import { umDynamicSettings } from './lib/saved_objects';
+import { mappingFromFieldMap } from '../../rule_registry/common/mapping_from_field_map';
+import { RuleDataClient } from '../../rule_registry/server';
+import { TECHNICAL_COMPONENT_TEMPLATE_NAME } from '../../rule_registry/common/assets';
+
+export type UptimeRuleRegistry = ReturnType<Plugin['setup']>['ruleRegistry'];
 
 export class Plugin implements PluginType {
   private savedObjectsClient?: ISavedObjectsRepository;
+  private initContext: PluginInitializerContext;
 
-  constructor(_initializerContext: PluginInitializerContext) {}
+  constructor(_initializerContext: PluginInitializerContext) {
+    this.initContext = _initializerContext;
+  }
 
   public setup(core: CoreSetup, plugins: UptimeCorePlugins) {
-    initServerWithKibana({ router: core.http.createRouter() }, plugins);
+    const getCoreStart = () => core.getStartServices().then(([coreStart]) => coreStart);
+
+    const { ruleDataService } = plugins.ruleRegistry;
+
+    const ready = once(async () => {
+      const componentTemplateName = ruleDataService.getFullAssetName('synthetics-mappings');
+
+      if (!ruleDataService.isWriteEnabled()) {
+        return;
+      }
+
+      await ruleDataService.createOrUpdateComponentTemplate({
+        name: componentTemplateName,
+        body: {
+          template: {
+            settings: {
+              number_of_shards: 1,
+            },
+            mappings: mappingFromFieldMap(uptimeRuleFieldMap),
+          },
+        },
+      });
+
+      await ruleDataService.createOrUpdateIndexTemplate({
+        name: ruleDataService.getFullAssetName('synthetics-index-template'),
+        body: {
+          index_patterns: [ruleDataService.getFullAssetName('observability-synthetics*')],
+          composed_of: [
+            ruleDataService.getFullAssetName(TECHNICAL_COMPONENT_TEMPLATE_NAME),
+            componentTemplateName,
+          ],
+        },
+      });
+    });
+
+    const ruleDataClient = new RuleDataClient({
+      alias: ruleDataService.getFullAssetName('observability-synthetics'),
+      getClusterClient: async () => {
+        const coreStart = await getCoreStart();
+        return coreStart.elasticsearch.client.asInternalUser;
+      },
+      ready,
+    });
```

Comment:
:information_source: elastic/kibana#103646 brought a small change in how the `RuleDataClient` should be instantiated. This is how I would propose to do it so the read-only feature flag works correctly:

```suggestion
    // initialize eagerly
    const initializeRuleDataTemplatesPromise = ready().catch(
      (err) => {
        this.logger!.error(err);
      }
    );

    const ruleDataClient = ruleDataService.getRuleDataClient(
      ruleDataService.getFullAssetName('observability-synthetics'),
      () => initializeRuleDataTemplatesPromise
    );
```

On an unrelated note, the `observability-synthetics` alias is probably not correct. The spec calls for `observability.synthetics` if I understand it correctly.

Reply:
Interesting. I see that apm is using the `observability-` alias as well. I see in the design document of course `observability.`. Can we confirm? @jasonrhodes